### PR TITLE
Device token is short

### LINKED ISSUE #99: deviceTokens have incorrect length
deviceToken bytes smaller than 10 doesn't get a trailing 0 and thus the deviceToken is too short.

### PR DESCRIPTION
**Pull request by ligerjohn**
Fixes #99

Incorrectly truncated values below 10. Fixed.
Incorrectly handled little/big endian. Fixed.

### CODE REVIEW

File: LigerTestApp/LigerTests/Startup/LGRAppDelegateTest.m [hunk 1]
```diff
@@ -43,6 +47,11 @@ - (void)testInit
 	XCTAssertNotNil([[LGRAppDelegate alloc] init], @"Failed to init");
 }
 
+- (void)testTestToken
+{
+	XCTAssertTrue([testTokenAsString() length] == 64, @"Test string should be 64 long");
```
Comment (tonyc) [inline LigerTestApp/LigerTests/Startup/LGRAppDelegateTest.m]: This is only testing that your hardcoded string is the correct length.

Are there tests around ```testToken``` itself to make sure the bytes come out in the correct order?

Comment (ligerjohn) [inline LigerTestApp/LigerTests/Startup/LGRAppDelegateTest.m]: Let me harden it up a bit.

Comment (ligerjohn) [inline LigerTestApp/LigerTests/Startup/LGRAppDelegateTest.m]: Added a test to verify endianness (effectively) as well as the length of the binary data.

File: LigerTestApp/LigerTests/Startup/LGRAppDelegateTest.m [hunk 2]
```diff
@@ -21,12 +21,16 @@ @interface LGRAppDelegateTest : XCTestCase
...
+	int32_t hexData[8] = {0x260a0f58, 0x99ac6bd8, 0xa3e0d6b5, 0x1f38a4ad, 0x3475c1e4, 0xeefbeee6, 0x2eca722c, 0xef0c3bf9};
+#elif __LITTLE_ENDIAN__
+	int32_t hexData[8] = {0x580f0a26, 0xd86bac99, 0xb5d6e0a3, 0xada4381f, 0xe4c17534, 0xe6eefbee, 0x2c72ca2e, 0xf93b0cef};
+#endif
 	return [NSData dataWithBytes:hexData length:32];
 }
 
 NSString* testTokenAsString() {
-	return @"26eaf5899ac6bd8a3e0d6b51f38a4ad3475c1e4eefbeee62eca722cefc3bf9";
+	return @"260a0f5899ac6bd8a3e0d6b51f38a4ad3475c1e4eefbeee62eca722cef0c3bf9";
```
Comment (tonyc) [inline LigerTestApp/LigerTests/Startup/LGRAppDelegateTest.m]: I still don't understand the value of this test...

File: LigerTestApp/LigerTests/Startup/LGRAppDelegateTest.m [hunk 3]
```diff
@@ -43,6 +47,30 @@ - (void)testInit
 	XCTAssertNotNil([[LGRAppDelegate alloc] init], @"Failed to init");
 }
 
+- (void)testTestTokenAsString
```
Comment (tonyc) [inline LigerTestApp/LigerTests/Startup/LGRAppDelegateTest.m]: I still don't understand the value of this test. You're testing hardcoded test data, not application logic.

Comment (ligerjohn) [inline LigerTestApp/LigerTests/Startup/LGRAppDelegateTest.m]: It was added to verify that the test data is correct. The old test data proved to be incorrect, and thus resulted in tests passing when they should have failed.

```- (void)testDidRegisterForRemoteNotificationsWithDeviceToken``` uses ```testTokenAsString()``` to verify that the NSData being sent in is rendered as the string we know is correct.

**Comment by tonyc** [comment]
:+1:

**Pull request merged**